lib/webapi/webhandler.go

```diff
@@ -634,6 +651,21 @@ func getWebConfigAuthSettings(cfg WebHandlerConfig) telewebui.WebConfigAuthSetti
 	return authSettings
 }
 
+func (h *WebHandler) disabledUI() bool {
+	sites, err := h.cfg.Operator.GetSites(defaults.SystemAccountID)
```

Comment:
If I remember correctly, each call to GetSites does require an etcd read, so I think there might be a risk that unauthenticated access can cause high query load on etcd. So similar to Dima, I would consider an approach where this is configured when setting up the handler as opposed to when the various handlers are used.